data/src/main/java/com/mashup/gabbangzip/sharedalbum/data/repository/LoginRepositoryImpl.kt

```diff
@@ -0,0 +1,39 @@
+package com.mashup.gabbangzip.sharedalbum.data.repository
+
+import com.mashup.gabbangzip.sharedalbum.data.dto.request.LoginRequest
+import com.mashup.gabbangzip.sharedalbum.data.service.LoginService
+import com.mashup.gabbangzip.sharedalbum.domain.datasource.LocalDataSource
+import com.mashup.gabbangzip.sharedalbum.domain.model.LoginParam
+import com.mashup.gabbangzip.sharedalbum.domain.repository.LoginRepository
+import javax.inject.Inject
+
+class LoginRepositoryImpl @Inject constructor(
+    private val loginService: LoginService,
+    private val localDataSource: LocalDataSource,
+) : LoginRepository {
+    override suspend fun login(param: LoginParam) {
+        val request = LoginRequest(
+            idToken = param.idToken,
+            provider = param.provider,
+            nickname = param.nickname,
+            profileImage = param.profileImage,
+        )
+        runCatching {
+            loginService.login(loginRequest = request)
+        }.onSuccess { response ->
+            with(response.data!!) {
+                saveToken(
+                    accessToken = accessToken,
+                    refreshToken = refreshToken,
+                )
+            }
```

Comment:
```suggestion
            response.data?.run {
                saveToken(
                    accessToken = accessToken,
                    refreshToken = refreshToken,
                )
            } ?: Exception("null response") // Exception 처리를 원한다면 이런느낌으로?!
```
!! 이라는건 결국 response.data 가 nullable 할 수 있다는 것인데, 방어코드 차원에서 null check 를 하거나 null 일 경우의 Exception 처리를 원한다면 Exception 을 보내주는 것은 어떨까?!

Reply:
API 요청에 실패했을 때 data가 null이라서 그때도 동일한 클래스를 사용하니까 nullable이라고 생각했는데 그냥 null을 보낼 수도 있겠넹 수정할겡

Reply:
[36a5294](https://github.com/mash-up-kr/gabbangzip-Android/pull/19/commits/36a5294c6547183e72741db8753366602524379f) 반영완!